### PR TITLE
Feat/zodiac guilds

### PR DESCRIPTION
**Pull request by fnanni-0**
#278

### CODE REVIEW

File: contracts/erc20guild/implementations/ZodiacERC20Guild.sol [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+pragma solidity ^0.8.17;
+
+import "@gnosis.pm/zodiac/contracts/interfaces/IAvatar.sol";
+import "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+import "@openzeppelin/contracts-upgradeable/utils/math/SafeMathUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/ECDSAUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/interfaces/IERC1271Upgradeable.sol";
+import "../ERC20GuildUpgradeable.sol";
+
+interface IPermissionRegistry {
```
Comment (AugustoL) [inline contracts/erc20guild/implementations/ZodiacERC20Guild.sol:11]: Lets create an interface file for this.

File: contracts/erc20guild/implementations/ZodiacERC20Guild.sol [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    // solhint-disable-next-line max-line-length
+    /// @param _votingPowerPercentageForProposalExecution The percentage of voting power in base 10000 needed to execute a proposal option
+    // solhint-disable-next-line max-line-length
+    /// @param _votingPowerPercentageForProposalCreation The percentage of voting power in base 10000 needed to create a proposal
+    /// @param _voteGas The amount of gas in wei unit used for vote refunds.
+    // Can't be higher than the gas used by setVote (117000)
+    /// @param _maxGasPrice The maximum gas price used for vote refunds
+    /// @param _maxActiveProposals The maximum amount of proposals to be active at the same time
+    /// @param _lockTime The minimum amount of seconds that the tokens would be locked
+    function setConfig(
```
Comment (AugustoL) [inline contracts/erc20guild/implementations/ZodiacERC20Guild.sol:51]: why do we have to override this?

File: contracts/erc20guild/implementations/ZodiacERC20Guild.sol [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+        minimumMembersForProposalCreation = _minimumMembersForProposalCreation;
+        minimumTokensLockedForProposalCreation = _minimumTokensLockedForProposalCreation;
+    }
+
+    /// @dev Set the ERC20Guild module configuration, can be called only executing a proposal or when it is initialized.
+    /// @param _avatar Address that this module will pass transactions to.
+    /// @param _multisend Address of the multisend contract that the avatar contract should use to bundle transactions.
+    function setAvatar(address _avatar, address _multisend) external virtual {
+        require(
+            msg.sender == address(this) || (msg.sender == avatar && isExecutingProposal),
```
Comment (AugustoL) [inline contracts/erc20guild/implementations/ZodiacERC20Guild.sol:91]: does it makes sense to check against `address(this)`?

Comment (fnanni-0) [reply contracts/erc20guild/implementations/ZodiacERC20Guild.sol:91]: No, but there should be a way of initially setting this up. I'll fix it.

**Review by AugustoL — APPROVED** [review]

**Pull request merged**